### ISSUE TITLE
Remove dependencies on other third-party Boot tasks

### ISSUE THREAD

**Issue by pandeiro** [pandeiro/boot-test-cljs#1]
## Rationale

I would like `boot-test-cljs` to be a fairly general-purpose task that can be used both standalone and within a "pipeline" of other boot tasks.

As it is currently implemented, the `test-cljs` task achieves the standalone use case, but does not compose well (or at all, if you try to run it with the `cljs` task as well). It will also cause transitive dependency issues if a project is creating its own sub-libraries that compose tasks using these third-party tools.
## Plan

Remove the parts that depend on `adzerk/boot-cljs` and `pandeiro/boot-http` and implement simplified versions of their functionality, keeping the API and functionality the same as it is, currently.

**Comment by myguidingstar-zz** [comment]
Yeah, shared tasks should only represent a specific step of build so they can be composed in project tasks.
Here's my implementation for nodejs runner.
https://github.com/boot-cljs-test/node-runner
Would you mind making a runner for full environments (phatomjs, browsers)?

**Comment by pandeiro** [comment]
@myguidingstar Sure I think I could port the HtmlUnit (browser) code over. Where is the implementation?

**Comment by myguidingstar-zz** [comment]
oh I forgot it. Link added above
